Tests/LeafKitTests/LeafKitTests.swift

```diff
@@ -916,21 +916,33 @@ final class LeafKitTests: XCTestCase {
             eventLoop: group.next()
         )
         
+        let progressLock = Lock()
+        var progress = 0
+        var done = false
+        var delay = 0
+        
         for iteration in 1...iterations {
-            let key = String((iteration % templates) + 1)
-            _ = renderer.render(path: key, context: [:])
+            let template = String((iteration % templates) + 1)
+            group.next()
+                .submit { renderer.render(path: template, context: [:]) }
+                .whenComplete { result in progressLock.withLock { progress += 1 } }
         }
         
-        group.shutdownGracefully { shutdown in
-            guard shutdown == nil else { XCTFail("ELG shutdown issue"); return }
-            XCTAssertEqual(renderer.cache.entryCount(), templates)
+        while !done {
+            progressLock.withLock { delay = (iterations - progress) * 10 }
+            guard delay == 0 else { usleep(UInt32(delay)); break }
+            done = true
+            group.shutdownGracefully { shutdown in
```

Comment:
Generally in tests we just do `try group.syncShutdownGracefully()` rather than dealing with the problem of making sure an async shutdown is complete before the test returns.